[Wei Nian] iP

hweinian opened the pull request:
# Gravitas

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Gravitas frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)
#
If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```java


public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

hweinian committed:
Level-0: rename chatbot

hweinian committed:
level 1

hweinian committed:
level 2

hweinian committed:
level 3

hweinian committed:
Level 4

hweinian committed:
Add Automated Text UI Testing

hweinian committed:
Level 5

hweinian committed:
Level 6

hweinian committed:
Add saving and loading of data

hweinian committed:
Revert "Add saving and loading of data"

This reverts commit 745d770c36dbf9f405fcaa69e38b282a89257eb3.

hweinian committed:
Merge branch 'branch-Level-7'

hweinian committed:
Reapply "Add saving and loading of data"

This reverts commit e4281193dbdea0267c606ba54e40daff1636b982.

hweinian committed:
Merge branch 'branch-Level-7'

hweinian committed:
Add date and time

hweinian committed:
Merge branch 'branch-Level-8'

hweinian committed:
Add more OOP

hweinian committed:
Divide classes into packages

hweinian committed:
Merge tag 'A-Packages'

hweinian committed:
Change time type from string to localtime

hweinian committed:
Merge branch 'add-gradle-support'

hweinian committed:
Add JUnit test cases for parser and event class

hweinian committed:
Edit gradle file

hweinian committed:
Add java docs

hweinian committed:
Improve coding quality standard

hweinian committed:
Add find feature to search for keyword

hweinian committed:
Merge branch 'branch-A-JavaDoc'

hweinian committed:
Merge tag 'A-CodingStandard'

# Conflicts:
#	src/main/java/gravitas/ui/Ui.java

hweinian committed:
Merge branch 'branch-Level-9'

hweinian committed:
Add checkstyle feature

hweinian committed:
Merge branch 'branch-A-CheckStyle'

hweinian committed:
Add gui feature to gravitas

hweinian committed:
Merge branch 'branch-Level-10'

wassilililily commented on src/test/java/gravitas/task/EventTest.java:
```diff
@@ -0,0 +1,69 @@
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Maybe you can avoid using wildcard imports but rather specify the imported class

wassilililily commented on src/main/java/gravitas/command/AddCommand.java:
```diff
@@ -0,0 +1,68 @@
+package gravitas.command;
```
I like your naming of packages!

wassilililily commented on src/main/java/gravitas/task/Deadline.java:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    @Override
+    public String formatData() {
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("d/M/yyyy");
+        DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("HHmm");
+        String mark = this.isDone ? "1" : "0";
+        return (this.eventType + " | " + mark + " | " + this.description
+                + " | " + this.endDate.format(dateFormatter) + " | "
+                + this.endTime.format(timeFormatter) + "\n");
+    }
+}
```
Very neat and clean codes! Well done!

wassilililily commented on src/main/java/gravitas/command/ExitCommand.java:
```diff
@@ -0,0 +1,29 @@
+
+    private static final String BYE = "Bye. Hope to see you again soon!";
+
+    /**
+     * Constructor for the ExitCommand class.
+     */
```
I like how you have a consistent layout for your comments!

wassilililily commented on src/main/java/gravitas/command/ListTaskCommand.java:
```diff
@@ -0,0 +1,24 @@
+     * Executes the command to list all tasks in the task list.
+     */
+    @Override
+    public String execute(TaskList taskList, Storage storage) throws DukeException {
```
Maybe you can change the name of the method so that it is more specific such as 'executeCommand' but overall your code is really well written!

wassilililily reviewed (commented):
Looks good 👍

AlekKwek18 commented:
LGTM

AlekKwek18 commented on src/main/java/gravitas/Gravitas.java:
```diff
@@ -0,0 +1,52 @@
+public class Gravitas {
+
+    private Storage storage;
+    private TaskList tasks;
```
Correct use of variable tasks when naming variables that stores more than one task

AlekKwek18 commented on src/main/java/gravitas/command/FindCommand.java:
```diff
@@ -0,0 +1,37 @@
+        this.userInput = userInput;
+    }
+
+    /**
```
for javadocs, explain what does string return i.e @return returns the output of all the lists find by the task list

AlekKwek18 commented on src/main/java/gravitas/command/UpdateTaskCommand.java:
```diff
@@ -0,0 +1,50 @@
+ */
+public class UpdateTaskCommand extends Command {
+
+    private static final String OUT_OF_BOUND = "The task that you wish to mark is invalid! please try again!";
```
Just a suggestion to rename OUT_OF_BOUND as INVALID_TASK_MESSAGE. MARK as TASK_MARKED_MESSAGE and UNMARK as TASK_UNMARKED_MESSAGE for betting naming convention

hweinian committed:
Add java assertion feature to validate assumptions within code

The codebase lacks checks to ensure assumptions and invariants to hold during runtime.
Runtime issues can be flagged and debugging runtime bugs will be easier.

Assertions is added throughout the code to verify assumptions and to flag issues.

These assumptions can be enabled during developement, and disabled during production to avoid performance overhead.

Assertions should be enabled during developement and testing but disabled for production builds to avoid runtime cost.

hweinian committed:
Improve code quality

Methods and variable names are not representative of what the actual method and variables do
which may lead to bug inducing code due to misunderstanding of what the variable actually do.

Method names and variable names are renamed to better reflect their functionality and purpose.
Group static variable together with common prefixes

Changes was done to ensure anyone working on the project is able to understand what is written with little to no ambiguity

hweinian committed:
Merge pull request #1 from hweinian/branch-A-Assertions

Add java assertion feature to validate assumptions within code

hweinian committed:
Merge pull request #2 from hweinian/master

Sync master branch to code quality

hweinian committed:
Merge pull request #3 from hweinian/branch-A-CodeQuality

Merge code quality branch

hweinian committed:
Add help command to improve user guidance

The app lacks a dedicated help command, which makes it difficult for users to understand how to use the GUI

To address this, a help command displays the possible feature that could be done by the app.

It is implemented in the GUI as users may want to know what command is available for them at any point in time

hweinian committed:
Merge branch 'branch-BCD-Extension'

hweinian committed:
Update DukeException name to GravitasException

hweinian committed:
Revert "Update DukeException name to GravitasException"

This reverts commit bed9a390ce99f9eb046395a90c5c69551b69a2ac.

hweinian committed:
Update DukeException to GravitasException

hweinian committed:
Add a user guide

hweinian committed:
Fix add command bugs pertaining to whitespaces

hweinian committed:
Update file path

hweinian committed:
Update file path of storage to relative file path

hweinian committed:
Fix bug related to the adding task
